pkg/util/kubeobjects/internal/query/query.go

```diff
@@ -56,6 +56,13 @@ func (c Generic[T, L]) Create(ctx context.Context, object T) error {
 func (c Generic[T, L]) Update(ctx context.Context, object T) error {
 	c.log(object).Info("updating")
 
+	if c.Owner != nil {
```

Comment:
just curios: can we write a test for it ?

Reply:
done